compiler/rustc_codegen_llvm/src/gotoc/rvalue.rs

```diff
@@ -926,8 +926,8 @@ impl<'tcx> GotocCtx<'tcx> {
         }
 
         // The source destination must be a fat pointers to a dyn trait object
-        assert!(self.is_vtable_fat_pointer(src_mir_type));
-        assert!(self.is_vtable_fat_pointer(dst_mir_type));
+        assert!(self.is_ref_of_unsized(src_mir_type));
```

Comment:
As above - would this allow slice pointer through?

Reply:
Fixed!